<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.1//EN" "http://www.w3.org/TR/xhtml11/DTD/xhtml11.dtd">
<html xmlns="http://www.w3.org/1999/xhtml"><head><title>TestNG Report</title><style type="text/css">table {margin-bottom:10px;border-collapse:collapse;empty-cells:show}th,td {border:1px solid #009;padding:.25em .5em}th {vertical-align:bottom}td {vertical-align:top}table a {font-weight:bold}.stripe td {background-color: #E6EBF9}.num {text-align:right}.passedodd td {background-color: #3F3}.passedeven td {background-color: #0A0}.skippedodd td {background-color: #DDD}.skippedeven td {background-color: #CCC}.failedodd td,.attn {background-color: #F33}.failedeven td,.stripe .attn {background-color: #D00}.stacktrace {white-space:pre;font-family:monospace}.totop {font-size:85%;text-align:center;border-bottom:2px solid #000}</style></head><body><table><tr><th>Test</th><th># Passed</th><th># Skipped</th><th># Failed</th><th>Time (ms)</th><th>Included Groups</th><th>Excluded Groups</th></tr><tr><th colspan="7">Default suite</th></tr><tr><td><a href="#t0">Default test</a></td><td class="num">9</td><td class="num">0</td><td class="num">0</td><td class="num">2,748</td><td></td><td></td></tr></table><table><thead><tr><th>Class</th><th>Method</th><th>Start</th><th>Time (ms)</th></tr></thead><tbody><tr><th colspan="4">Default suite</th></tr></tbody><tbody id="t0"><tr><th colspan="4">Default test &#8212; passed</th></tr><tr class="passedeven"><td rowspan="9">com.epam.apartment.dao.UserDaoTest</td><td><a href="#m0">authoriseUserReturnNullTest</a></td><td rowspan="1">1492179784170</td><td rowspan="1">48</td></tr><tr class="passedeven"><td><a href="#m1">authoriseUserSuccessTest</a></td><td rowspan="1">1492179784247</td><td rowspan="1">11</td></tr><tr class="passedeven"><td><a href="#m2">changePswdSuccessTest</a></td><td rowspan="1">1492179784290</td><td rowspan="1">15</td></tr><tr class="passedeven"><td><a href="#m3">editProfileSuccessTest</a></td><td rowspan="1">1492179784341</td><td rowspan="1">34</td></tr><tr class="passedeven"><td><a href="#m4">editProfileThrowsDuplicatedKeyExceptionTest</a></td><td rowspan="1">1492179784461</td><td rowspan="1">198</td></tr><tr class="passedeven"><td><a href="#m5">registerNewUserSuccessTest</a></td><td rowspan="1">1492179784751</td><td rowspan="1">14</td></tr><tr class="passedeven"><td><a href="#m6">registerNewUserWithSameEmailTest</a></td><td rowspan="1">1492179784962</td><td rowspan="1">6</td></tr><tr class="passedeven"><td><a href="#m7">restorePswdSuccessTest</a></td><td rowspan="1">1492179785026</td><td rowspan="1">7</td></tr><tr class="passedeven"><td><a href="#m8">restorePswdWithNotFoundEmailTest</a></td><td rowspan="1">1492179785066</td><td rowspan="1">3</td></tr></tbody></table><h2>Default test</h2><h3 id="m0">com.epam.apartment.dao.UserDaoTest#authoriseUserReturnNullTest</h3><table class="result"></table><p class="totop"><a href="#summary">back to summary</a></p><h3 id="m1">com.epam.apartment.dao.UserDaoTest#authoriseUserSuccessTest</h3><table class="result"></table><p class="totop"><a href="#summary">back to summary</a></p><h3 id="m2">com.epam.apartment.dao.UserDaoTest#changePswdSuccessTest</h3><table class="result"></table><p class="totop"><a href="#summary">back to summary</a></p><h3 id="m3">com.epam.apartment.dao.UserDaoTest#editProfileSuccessTest</h3><table class="result"></table><p class="totop"><a href="#summary">back to summary</a></p><h3 id="m4">com.epam.apartment.dao.UserDaoTest#editProfileThrowsDuplicatedKeyExceptionTest</h3><table class="result"><tr><th>Expected Exception</th></tr><tr><td><div class="stacktrace">org.springframework.dao.DuplicateKeyException: PreparedStatementCallback; SQL [UPDATE USERS SET U_EMAIL = ?, U_NAME = ?, U_SURNAME = ? WHERE U_ID = ?]; Unique index or primary key violation: &quot;U_EMAIL_UNIQUE_INDEX_4 ON PUBLIC.USERS(U_EMAIL) VALUES (&apos;alesnax@gmail.com&apos;, 1)&quot;; SQL statement:
UPDATE USERS SET U_EMAIL = ?, U_NAME = ?, U_SURNAME = ? WHERE U_ID = ? [23505-193]; nested exception is org.h2.jdbc.JdbcSQLException: Unique index or primary key violation: &quot;U_EMAIL_UNIQUE_INDEX_4 ON PUBLIC.USERS(U_EMAIL) VALUES (&apos;alesnax@gmail.com&apos;, 1)&quot;; SQL statement:
UPDATE USERS SET U_EMAIL = ?, U_NAME = ?, U_SURNAME = ? WHERE U_ID = ? [23505-193]
	at org.springframework.jdbc.support.SQLErrorCodeSQLExceptionTranslator.doTranslate(SQLErrorCodeSQLExceptionTranslator.java:239)
	at org.springframework.jdbc.support.AbstractFallbackSQLExceptionTranslator.translate(AbstractFallbackSQLExceptionTranslator.java:73)
	at org.springframework.jdbc.core.JdbcTemplate.execute(JdbcTemplate.java:649)
	at org.springframework.jdbc.core.JdbcTemplate.update(JdbcTemplate.java:870)
	at org.springframework.jdbc.core.JdbcTemplate.update(JdbcTemplate.java:894)
	at org.springframework.jdbc.core.namedparam.NamedParameterJdbcTemplate.update(NamedParameterJdbcTemplate.java:287)
	at com.epam.apartment.dao.impl.JdbcUserDao.editProfile(JdbcUserDao.java:108)
	at com.epam.apartment.dao.UserDaoTest.editProfileThrowsDuplicatedKeyExceptionTest(UserDaoTest.java:125)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
	at java.lang.reflect.Method.invoke(Unknown Source)
	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:85)
	at org.testng.internal.MethodInvocationHelper$1.runTestMethod(MethodInvocationHelper.java:200)
	at org.springframework.test.context.testng.AbstractTestNGSpringContextTests.run(AbstractTestNGSpringContextTests.java:175)
	at org.testng.internal.MethodInvocationHelper.invokeHookable(MethodInvocationHelper.java:212)
	at org.testng.internal.Invoker.invokeMethod(Invoker.java:652)
	at org.testng.internal.Invoker.invokeTestMethod(Invoker.java:845)
	at org.testng.internal.Invoker.invokeTestMethods(Invoker.java:1153)
	at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:125)
	at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:108)
	at org.testng.TestRunner.privateRun(TestRunner.java:771)
	at org.testng.TestRunner.run(TestRunner.java:621)
	at org.testng.SuiteRunner.runTest(SuiteRunner.java:357)
	at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:352)
	at org.testng.SuiteRunner.privateRun(SuiteRunner.java:310)
	at org.testng.SuiteRunner.run(SuiteRunner.java:259)
	at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:52)
	at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:86)
	at org.testng.TestNG.runSuitesSequentially(TestNG.java:1199)
	at org.testng.TestNG.runSuitesLocally(TestNG.java:1124)
	at org.testng.TestNG.run(TestNG.java:1032)
	at org.testng.remote.AbstractRemoteTestNG.run(AbstractRemoteTestNG.java:132)
	at org.testng.remote.RemoteTestNG.initAndRun(RemoteTestNG.java:230)
	at org.testng.remote.RemoteTestNG.main(RemoteTestNG.java:76)
Caused by: org.h2.jdbc.JdbcSQLException: Unique index or primary key violation: &quot;U_EMAIL_UNIQUE_INDEX_4 ON PUBLIC.USERS(U_EMAIL) VALUES (&apos;alesnax@gmail.com&apos;, 1)&quot;; SQL statement:
UPDATE USERS SET U_EMAIL = ?, U_NAME = ?, U_SURNAME = ? WHERE U_ID = ? [23505-193]
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:345)
	at org.h2.message.DbException.get(DbException.java:179)
	at org.h2.message.DbException.get(DbException.java:155)
	at org.h2.index.BaseIndex.getDuplicateKeyException(BaseIndex.java:103)
	at org.h2.mvstore.db.MVSecondaryIndex.checkUnique(MVSecondaryIndex.java:231)
	at org.h2.mvstore.db.MVSecondaryIndex.add(MVSecondaryIndex.java:190)
	at org.h2.mvstore.db.MVTable.addRow(MVTable.java:704)
	at org.h2.table.Table.updateRows(Table.java:508)
	at org.h2.command.dml.Update.update(Update.java:145)
	at org.h2.command.CommandContainer.update(CommandContainer.java:98)
	at org.h2.command.Command.executeUpdate(Command.java:258)
	at org.h2.jdbc.JdbcPreparedStatement.executeUpdateInternal(JdbcPreparedStatement.java:160)
	at org.h2.jdbc.JdbcPreparedStatement.executeUpdate(JdbcPreparedStatement.java:146)
	at org.springframework.jdbc.core.JdbcTemplate$2.doInPreparedStatement(JdbcTemplate.java:877)
	at org.springframework.jdbc.core.JdbcTemplate$2.doInPreparedStatement(JdbcTemplate.java:870)
	at org.springframework.jdbc.core.JdbcTemplate.execute(JdbcTemplate.java:633)
	... 32 more
</div></td></tr></table><p class="totop"><a href="#summary">back to summary</a></p><h3 id="m5">com.epam.apartment.dao.UserDaoTest#registerNewUserSuccessTest</h3><table class="result"></table><p class="totop"><a href="#summary">back to summary</a></p><h3 id="m6">com.epam.apartment.dao.UserDaoTest#registerNewUserWithSameEmailTest</h3><table class="result"><tr><th>Expected Exception</th></tr><tr><td><div class="stacktrace">org.springframework.dao.DuplicateKeyException: PreparedStatementCallback; SQL [INSERT INTO USERS (U_EMAIL, U_NAME, U_SURNAME, U_PASSWORD, U_BIRTHDAY) VALUES (?, ?, ?, ?, ?)]; Unique index or primary key violation: &quot;U_EMAIL_UNIQUE_INDEX_4 ON PUBLIC.USERS(U_EMAIL) VALUES (&apos;qnewemail@gmail.com&apos;, 14)&quot;; SQL statement:
INSERT INTO USERS (U_EMAIL, U_NAME, U_SURNAME, U_PASSWORD, U_BIRTHDAY) VALUES (?, ?, ?, ?, ?) [23505-193]; nested exception is org.h2.jdbc.JdbcSQLException: Unique index or primary key violation: &quot;U_EMAIL_UNIQUE_INDEX_4 ON PUBLIC.USERS(U_EMAIL) VALUES (&apos;qnewemail@gmail.com&apos;, 14)&quot;; SQL statement:
INSERT INTO USERS (U_EMAIL, U_NAME, U_SURNAME, U_PASSWORD, U_BIRTHDAY) VALUES (?, ?, ?, ?, ?) [23505-193]
	at org.springframework.jdbc.support.SQLErrorCodeSQLExceptionTranslator.doTranslate(SQLErrorCodeSQLExceptionTranslator.java:239)
	at org.springframework.jdbc.support.AbstractFallbackSQLExceptionTranslator.translate(AbstractFallbackSQLExceptionTranslator.java:73)
	at org.springframework.jdbc.core.JdbcTemplate.execute(JdbcTemplate.java:649)
	at org.springframework.jdbc.core.JdbcTemplate.update(JdbcTemplate.java:870)
	at org.springframework.jdbc.core.JdbcTemplate.update(JdbcTemplate.java:894)
	at org.springframework.jdbc.core.namedparam.NamedParameterJdbcTemplate.update(NamedParameterJdbcTemplate.java:287)
	at com.epam.apartment.dao.impl.JdbcUserDao.registerNewUser(JdbcUserDao.java:78)
	at com.epam.apartment.dao.UserDaoTest.registerNewUserWithSameEmailTest(UserDaoTest.java:84)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
	at java.lang.reflect.Method.invoke(Unknown Source)
	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:85)
	at org.testng.internal.MethodInvocationHelper$1.runTestMethod(MethodInvocationHelper.java:200)
	at org.springframework.test.context.testng.AbstractTestNGSpringContextTests.run(AbstractTestNGSpringContextTests.java:175)
	at org.testng.internal.MethodInvocationHelper.invokeHookable(MethodInvocationHelper.java:212)
	at org.testng.internal.Invoker.invokeMethod(Invoker.java:652)
	at org.testng.internal.Invoker.invokeTestMethod(Invoker.java:845)
	at org.testng.internal.Invoker.invokeTestMethods(Invoker.java:1153)
	at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:125)
	at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:108)
	at org.testng.TestRunner.privateRun(TestRunner.java:771)
	at org.testng.TestRunner.run(TestRunner.java:621)
	at org.testng.SuiteRunner.runTest(SuiteRunner.java:357)
	at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:352)
	at org.testng.SuiteRunner.privateRun(SuiteRunner.java:310)
	at org.testng.SuiteRunner.run(SuiteRunner.java:259)
	at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:52)
	at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:86)
	at org.testng.TestNG.runSuitesSequentially(TestNG.java:1199)
	at org.testng.TestNG.runSuitesLocally(TestNG.java:1124)
	at org.testng.TestNG.run(TestNG.java:1032)
	at org.testng.remote.AbstractRemoteTestNG.run(AbstractRemoteTestNG.java:132)
	at org.testng.remote.RemoteTestNG.initAndRun(RemoteTestNG.java:230)
	at org.testng.remote.RemoteTestNG.main(RemoteTestNG.java:76)
Caused by: org.h2.jdbc.JdbcSQLException: Unique index or primary key violation: &quot;U_EMAIL_UNIQUE_INDEX_4 ON PUBLIC.USERS(U_EMAIL) VALUES (&apos;qnewemail@gmail.com&apos;, 14)&quot;; SQL statement:
INSERT INTO USERS (U_EMAIL, U_NAME, U_SURNAME, U_PASSWORD, U_BIRTHDAY) VALUES (?, ?, ?, ?, ?) [23505-193]
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:345)
	at org.h2.message.DbException.get(DbException.java:179)
	at org.h2.message.DbException.get(DbException.java:155)
	at org.h2.index.BaseIndex.getDuplicateKeyException(BaseIndex.java:103)
	at org.h2.mvstore.db.MVSecondaryIndex.checkUnique(MVSecondaryIndex.java:231)
	at org.h2.mvstore.db.MVSecondaryIndex.add(MVSecondaryIndex.java:190)
	at org.h2.mvstore.db.MVTable.addRow(MVTable.java:704)
	at org.h2.command.dml.Insert.insertRows(Insert.java:156)
	at org.h2.command.dml.Insert.update(Insert.java:114)
	at org.h2.command.CommandContainer.update(CommandContainer.java:98)
	at org.h2.command.Command.executeUpdate(Command.java:258)
	at org.h2.jdbc.JdbcPreparedStatement.executeUpdateInternal(JdbcPreparedStatement.java:160)
	at org.h2.jdbc.JdbcPreparedStatement.executeUpdate(JdbcPreparedStatement.java:146)
	at org.springframework.jdbc.core.JdbcTemplate$2.doInPreparedStatement(JdbcTemplate.java:877)
	at org.springframework.jdbc.core.JdbcTemplate$2.doInPreparedStatement(JdbcTemplate.java:870)
	at org.springframework.jdbc.core.JdbcTemplate.execute(JdbcTemplate.java:633)
	... 32 more
</div></td></tr></table><p class="totop"><a href="#summary">back to summary</a></p><h3 id="m7">com.epam.apartment.dao.UserDaoTest#restorePswdSuccessTest</h3><table class="result"></table><p class="totop"><a href="#summary">back to summary</a></p><h3 id="m8">com.epam.apartment.dao.UserDaoTest#restorePswdWithNotFoundEmailTest</h3><table class="result"></table><p class="totop"><a href="#summary">back to summary</a></p></body></html>